File: clients/sellingpartner-api-aa-csharp/src/Amazon.SellingPartnerAPIAA/LWAException.cs
using System;
namespace Amazon.SellingPartnerAPIAA
{
	public class LWAException:Exception
	{
        private String errorMessage;

        private String errorCode;

        public LWAException():base()
        {

        }

        public LWAException(String errorCode, String errorMessage)
        {
            this.errorCode = errorCode;
            this.errorMessage = errorMessage;
        }

        public LWAException(String errorCode, String errorMessage, String message): base(message)
        {
            this.errorCode = errorCode;
            this.errorMessage = errorMessage;
        }

        public LWAException(String errorCode, String errorMessage, String message, Exception innerException) : base(message,innerException)
        {
            this.errorCode = errorCode;
            this.errorMessage = errorMessage;
        }


        public String getErrorCode()
        {
            return this.errorCode;
        }

        public String getErrorMessage()
        {
            return this.errorMessage;
        }

    }
}

File: clients/sellingpartner-api-aa-java/src/com/amazon/SellingPartnerAPIAA/LWAException.java
package com.amazon.SellingPartnerAPIAA;

public class LWAException extends Exception {

    private String errorMessage;

    private String errorCode;

    LWAException() {
        super();
    }

    LWAException(String errorCode, String errorMessage) {
        super();
        this.errorCode = errorCode;
        this.errorMessage = errorMessage;
    }

    LWAException(String errorCode, String errorMessage, String message) {
        super(message);
        this.errorCode = errorCode;
        this.errorMessage = errorMessage;
    }

    LWAException(String errorCode, String errorMessage, Throwable cause) {
        super(cause);
        this.errorCode = errorCode;
        this.errorMessage = errorMessage;
    }

    LWAException(String errorCode, String errorMessage, String message, Throwable cause) {
        super(message, cause);
        this.errorCode = errorCode;
        this.errorMessage = errorMessage;
    }

    public String getErrorCode() {
        return this.errorCode;
    }

    public String getErrorMessage() {
        return this.errorMessage;
    }

}
